Sources/OpenAPIRuntime/Errors/ClientError.swift

```diff
@@ -35,7 +40,13 @@ public struct ClientError: Error {
     ///
     /// Will be nil if the error resulted before the request was generated,
     /// for example if generating the request from the Input failed.
-    public var request: Request?
+    public var request: HTTPRequest?
+
+    /// The HTTP request body created during the operation.
+    ///
+    /// Will be nil if the error resulted before the request was generated,
+    /// for example if generating the request from the Input failed.
+    public var requestBody: HTTPBody?
```

Comment:
I don't remember the details, but in some reviews of the API the feedback was to avoid stored tuples, because what you suggest is what was originally proposed. The downside of the tuple approach is that the access requires the unfortunate `.0` and `.1`, or giving names to the tuple elements, which means the naming itself becomes tricky.

The best I could come up with would be:
```
public var requestInfo: (request: HTTPRequest, body: HTTPBody?)?
```

Which works, but I don't find it better than what we have today.

Regarding the optionality of the body, yes the body can be nil while the request is non-nil, for example for GET requests the body is never non-nil.

Reply:
Discussed more offline, keeping as separate properties for now.